pinot-minion/src/main/java/org/apache/pinot/minion/taskfactory/TaskFactoryRegistry.java

```diff
@@ -85,6 +89,11 @@ public TaskResult run() {
                   .addPhaseTiming(taskType, MinionQueryPhase.TASK_QUEUEING, jobDequeueTimeMs - jobInQueueTimeMs,
                       TimeUnit.MILLISECONDS);
               try {
+                if (separateTaskLogs) {
+                  LOGGER.info("Routing logs to separate file for task: {}", _taskConfig.getId());
+                  // Setting MDC so that logger can route task logs to a separate file.
+                  MDC.put("taskId", _taskConfig.getId());
```

Comment:
if all tasks running put/remove the same key "taskId" how does it know to differentiate and not overwrite?

Reply:
good question, this will run on its own thread, as you see it's in a closure to be called later
```
TaskFactory taskFactory = context -> {
```